testsuite/module-publish/src/packages/framework_usecases/sources/resource_groups_example.move

```diff
@@ -0,0 +1,201 @@
+                move_to(owner, resource);
+            }
+        } else if (index == 5) {
+            if (exists<ExampleResource5>(owner_address)) {
+                let resource = borrow_global_mut<ExampleResource5>(owner_address);
+                resource.name = name;
+            } else {
+                let resource = ExampleResource5 {
+                    value: 0,
+                    name,
+                };
+                move_to(owner, resource);
+            }
+        } else if (index == 6) {
+            if (exists<ExampleResource6>(owner_address)) {
+                let resource = borrow_global_mut<ExampleResource6>(owner_address);
+                resource.name = name;
+            } else {
+                let resource = ExampleResource6 {
+                    value: 0,
+                    name,
+                };
+                move_to(owner, resource);
+            }
+        } else if (index == 7) {
+            if (exists<ExampleResource7>(owner_address)) {
+                let resource = borrow_global_mut<ExampleResource7>(owner_address);
+                resource.name = name;
+            } else {
+                let resource = ExampleResource7 {
+                    value: 0,
+                    name,
+                };
+                move_to(owner, resource);
+            }
+        };
+    }
+
+    public entry fun set_p(_fee_payer: &signer, owner: &signer, index: u64, name: String) acquires ExampleResource0, ExampleResource1, ExampleResource2, ExampleResource3, ExampleResource4, ExampleResource5, ExampleResource6, ExampleResource7 {
+        set(owner, index, name);
+    }
+
+    public entry fun set_4(owner: &signer, index1: u64, index2: u64, index3: u64, index4: u64, name: String) acquires ExampleResource0, ExampleResource1, ExampleResource2, ExampleResource3, ExampleResource4, ExampleResource5, ExampleResource6, ExampleResource7 {
+        set(owner, index1, name);
+        set(owner, index2, name);
+        set(owner, index3, name);
+        set(owner, index4, name);
+    }
+
+    public entry fun read(_: &signer, owner: &signer, index: u64) acquires ExampleResource0 {
+        let owner_address = signer::address_of(owner);
+        if (index == 0) {
+            if (exists<ExampleResource0>(owner_address)) {
+                let resource = borrow_global_mut<ExampleResource0>(owner_address);
+                assert!(resource.value < 1000000000000, error::invalid_state(EVALUE_TOO_LARGE));
+            } else {
+                let resource = ExampleResource0 {
+                    value: 0,
+                    name: string::utf8(b"init_name"),
+                };
+                move_to(owner, resource);
+            }
+        } else {
+
+        }
+    }
+
+    #[test(owner = @0x123)]
+    entry fun test_set(owner: &signer) acquires Collection {
+        set(owner, owner, 0, string::utf8(b"a"));
+        set(owner, owner, 0, string::utf8(b"b"));
+        set(owner, owner, 1, string::utf8(b"c"));
```

Comment:
As this is a test, we should do some assertions that the change is actually performed right?

Reply:
set function also takes 3 arguments instead of 4 arguments.